### ISSUE TITLE
Robust downloading

### ISSUE THREAD

**Issue by interob** [WFP-VAM/modape#138]
Multi-threaded downloading seems to sometimes render files corrupt. In the download endpoint, a checksum is available so integrity could be checked. Requested enhancement of the downloading functionality:
1. Check endpoint availability
2. Download to a _temporary_ filename in the targeted location
3. When done, check file size (correct file size implies that the file exists ;-) )
4. Compute and check the hash / checksum
5. When all of the above is successful: rename the file to its actual filename

Step 4 could be optional; all others are cheap.

**Comment by interob** [comment]
And allowing another argument for the number of retries would be a big plus

**Comment by interob** [comment]
Tested the publicly available and "5 - Production/Stable" rated pycksum package: [https://pypi.org/project/pycksum/0.4.3/](https://pypi.org/project/pycksum/0.4.3/)
```
import pycksum
print(pycksum.cksum(open('MYD13A2.A2020329.h08v05.006.2020347035749.hdf', 'rb')))
```
`4043604692`

Computing the hash for a 17MB .hdf file takes about 100ms. When downloading and processing the full archive for the African continent @1km ( {42 MODIS tiles} x {period of mid-2002 to present} = 54k downloads) that would add 90 minutes to the otherwise 3 days of processing on an average AWS EC2 instance (running single threaded).

IMO, the benefit outweighs the costs.

**Comment by valpesendorfer** [comment]
Thanks @interob, I've had looked at this package as well so it's great that you've tested it.

**Comment by interob** [comment]
Also, consider this assertion:
https://github.com/WFP-VAM/modape/blob/5dd5fa7b942b0a6f6a946b353050d50b9abda9b2/modape/modis/download.py#L285
This sanity check fails every now and then; it's really needed that MODAPE can recover from such situation. Or maybe the more robust downloading we're planning to implement can generally prevent failure at this point.

**Comment by valpesendorfer** [comment]
implemented with #147
